qiita_pet/handlers/study_handlers/tests/test_artifact.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import main
+from os.path import exists, join
+from json import loads
+
+from qiita_pet.test.tornado_test_base import TestHandlerBase
+from qiita_db.artifact import Artifact
+from qiita_db.util import get_count, get_mountpoint
+from qiita_db.metadata_template.prep_template import PrepTemplate
+
+
+class ArtifactGraphAJAXTests(TestHandlerBase):
+    def test_get_ancestors(self):
+        response = self.get('/artifact/graph/', {'direction': 'ancestors',
+                                                 'artifact_id': 1})
+        exp = {'status': 'success',
+               'message': '',
+               'node_labels': [[1, 'Raw data 1 - FASTQ']],
+               'edge_list': []}
+        self.assertEqual(response.code, 200)
+        self.assertEqual(loads(response.body), exp)
+
+    def test_get_descendants(self):
+        response = self.get('/artifact/graph/', {'direction': 'descendants',
+                                                 'artifact_id': 1})
+        exp = {'status': 'success',
+               'message': '',
+               'node_labels': [[1, 'Raw data 1 - FASTQ'],
+                               [3, 'Demultiplexed 2 - Demultiplexed'],
+                               [2, 'Demultiplexed 1 - Demultiplexed'],
+                               [4, 'BIOM - BIOM'],
+                               [5, 'BIOM - BIOM']],
+               'edge_list': [[1, 3], [1, 2], [2, 4], [2, 5]]}
+        self.assertEqual(response.code, 200)
+        self.assertItemsEqual(loads(response.body), exp)
+
+    def test_get_unknown(self):
+        response = self.get('/artifact/graph/', {'direction': 'BAD',
+                                                 'artifact_id': 1})
+        exp = {'status': 'error',
+               'message': 'Unknown directon BAD'}
+        self.assertEqual(response.code, 200)
+        self.assertEqual(loads(response.body), exp)
+
+
+class NewArtifactHandlerTestsReadOnly(TestHandlerBase):
+    def test_get(self):
+        response = self.get('/study/add_prep/1')
+        self.assertEqual(response.code, 200)
+        self.assertIn('Select file type', response.body)
+        self.assertIn('uploaded_file.txt', response.body)
+
+
+class NewArtifactHandlerTests(TestHandlerBase):
+    database = True
+
+    def tearDown(self):
+        # Replace file if removed as part of function testing
+        uploads_path = get_mountpoint('uploads')[0][1]
+        fp = join(uploads_path, '1', 'uploaded_file.txt')
+        if not exists(fp):
+            with open(fp, 'w') as f:
+                f.write('')
+
+    def test_post_artifact(self):
+        new_artifact_id = get_count('qiita.artifact') + 1
```

Comment:
If we are moving towards the rollback test framework instead of the current wipe and rebuild the database, this will no longer work. You need to get the next ID in the series attached to the id column, as the IDs still increase regardless of the rollback.